src/lib/boxJava.js

```diff
@@ -101,3 +101,55 @@ try {
     }
   ]
 });
+
+foam.CLASS({
+  refines: 'foam.box.SessionReplyBox',
+
+  javaImports: [
+    'java.security.AccessControlException'
+  ],
+
+  properties: [
+    {
+      class: 'Object',
+      name: 'msg',
+      javaType: 'foam.box.Message'
+    },
+    {
+      class: 'Object',
+      name: 'clientBox',
+      javaType: 'foam.box.Box'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'send',
+      javaCode:
+`Object object = message.getObject();
+if ( object instanceof RPCErrorMessage && ((RPCErrorMessage) object).getData() instanceof AccessControlException ) {
+  // TODO: should this be wrapped in new Thread() ?
+  ((Runnable) getX().get("requestLogin")).run();
```

Comment:
@kgrgreer Should this be wrapped in a new thread?

Reply:
I don't think so, since you want to know when it's done and the easiest way to do that is to just block this thread until run() returns.

Reply:
Okay thanks.